app/src/common/shared/com/igalia/wolvic/telemetry/TelemetryService.java

```diff
@@ -175,7 +175,11 @@ public static void closeWindowEvent(int windowId) {
     }
 
     private static String getDefaultSearchEngineIdentifierForTelemetry() {
-        return SearchEngineWrapper.get(context).getCurrentSearchEngine().getIdentifier();
+        SearchEngineWrapper w = SearchEngineWrapper.get(context);
```

Comment:
Would it be possible to reuse this code by defining it in a static method ? If we eventually figure out a better approach we would just need to change such function. WDTY ?

Reply:
I'm not sure what you mean by saying a "static" method as clearly, it needs to be contextual here. Of course, we can have a method that includes these calls, but not simply modify getCurrentSearchEngine() as that will cause an infinite recursive call stack (initSearchEngine() will also need to getCurrentSearchEngine()).

Reply:
And I have just done so.